### ISSUE TITLE
`join_absolute_paths` lint generates incorrect fix

### ISSUE THREAD

**Issue by samueltardieu** [rust-lang/rust-clippy#16607]
### Summary

Some strings are not properly escaped in the suggestion.

### Reproducer

Code:

```rust
#![warn(clippy::join_abolute_paths)]

use std::path::{Path, PathBuf};

fn main() {
    let path = Path::new("C::\\Users");
    path.join("\\user");
}
```

suggests using `path.join("\user");` which is incorrect, as `\u` is the start of a unicode sequence which is not valid. Both "\\" should have been removed.

### Version

```text
rustc 1.95.0-nightly (0376d43d4 2026-02-20)
binary: rustc
commit-hash: 0376d43d443cba463a0b6a6ec9140ea17d7b7130
commit-date: 2026-02-20
host: x86_64-unknown-linux-gnu
release: 1.95.0-nightly
LLVM version: 22.1.0
```

### Additional Labels

Once this is fixed, the `//@no-rustfix` should be removed from `tests/ui/join_absolute_paths.rs`.
